### PR TITLE
[extractor/zattoo] Fix Zattoo resellers

### LINKED ISSUE #4630: [salttv] broken extractor (zattoo clone)
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a bug unrelated to a specific site
- [X] I've verified that I'm running yt-dlp version **2022.08.08** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/ytdl-org/youtube-dl#video-url-contains-an-ampersand-and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Provide a description that is worded well enough to be understood

First of all, thanks a lot to @goggle and team for the recent fixes, Zattoo looks fine now!   The issue here is about tv.salt.ch which is a rebranded zattoo.com: here the extractor doesn't seem to work at all (anymore). 

For a specific stream from today, zattoo download is working, Salt-TV (tv.salt.ch) is not.   For example for this stream it is ok with the Zattoo Extractor: 

```
$ yt-dlp --netrc https://zattoo.com/watch/telezueri/263168737
Zattoo] Downloading session token
[Zattoo] Opening session
[Zattoo] Logging in
[Zattoo] 263168737: Downloading video information (...)
```

but it fails with for the same show on Salt TV: 

```
$ yt-dlp --netrc https://tv.salt.ch/watch/telezueri/263052803
[SaltTV] Downloading session token
[SaltTV] Opening session
[SaltTV] Logging in
ERROR: This method must be implemented by subclasses
```

It is not a login issue (it then fails with a "Unable to login" message).      Not sure why this happens on SaltTv, as everything else seems to be the same as on the zattoo frontend. 

Do you have an idea how to fix this ?  I will also try 2-3 things the next days or weeks.  
Thanks & best regards. 

PS: Other small thing (possibly for another ticket), recorded shows extraction doesn't seem to work yet, or not always: 

```
$ yt-dlp --netrc https://tv.salt.ch/recording/463479573
[generic] 463479573: Downloading webpage
WARNING: [generic] Falling back on generic information extractor
[generic] 463479573: Extracting information
ERROR: Unsupported URL: https://tv.salt.ch/recording/463479573

$ yt-dlp --netrc https://zattoo.com/recording/444173552
[generic] 444173552: Downloading webpage
WARNING: [generic] Falling back on generic information extractor
[generic] 444173552: Extracting information
ERROR: Unsupported URL: https://zattoo.com/recording/444173552
```

but interesting enough, when switching to the another url scheme, it then works (stream from May 2022): 
```
yt-dlp --netrc https://zattoo.com/series/france-2/103863/2?recording=444173552
[ZattooRecordings] Downloading session token
[ZattooRecordings] Opening session
[ZattooRecordings] Logging in
[ZattooRecordings] 444173552: Downloading playlist
[ZattooRecordings] 254811212: Downloading video information (...)
```

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
$ yt-dlp -vU --netrc https://tv.salt.ch/watch/telezueri/263052803
[debug] Command-line config: ['-vU', '--netrc', 'https://tv.salt.ch/watch/telezueri/263052803']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.08.08 [3157158] (pip)
[debug] Python 3.10.6 (CPython 64bit) - macOS-12.5-x86_64-i386-64bit 
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: ffprobe -bsfs
[debug] exe versions: ffmpeg 5.1 (setts), ffprobe 5.1, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.15.0, brotli-1.0.9, certifi-2022.06.15, mutagen-1.45.1, sqlite3-2.6.0, websockets-10.3
[debug] Proxy map: {}
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2022.08.08, Current version: 2022.08.08
yt-dlp is up to date (2022.08.08)
[SaltTV] Downloading session token
[SaltTV] Opening session
[SaltTV] Logging in
[debug] [SaltTV] Extracting URL: https://tv.salt.ch/watch/telezueri/263052803
ERROR: This method must be implemented by subclasses
Traceback (most recent call last):
  File "/usr/local/Cellar/yt-dlp/2022.8.8/libexec/lib/python3.10/site-packages/yt_dlp/YoutubeDL.py", line 1441, in wrapper
    return func(self, *args, **kwargs)
  File "/usr/local/Cellar/yt-dlp/2022.8.8/libexec/lib/python3.10/site-packages/yt_dlp/YoutubeDL.py", line 1517, in __extract_info
    ie_result = ie.extract(url)
  File "/usr/local/Cellar/yt-dlp/2022.8.8/libexec/lib/python3.10/site-packages/yt_dlp/extractor/common.py", line 666, in extract
    ie_result = self._real_extract(url)
  File "/usr/local/Cellar/yt-dlp/2022.8.8/libexec/lib/python3.10/site-packages/yt_dlp/extractor/common.py", line 739, in _real_extract
    raise NotImplementedError('This method must be implemented by subclasses')
NotImplementedError: This method must be implemented by subclasses
```

### PR DESCRIPTION
**Pull request by goggle**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

</details>

This is an attempt to finally fix the `Zattoo` resellers.
My proposal is currently only done for `SaltTV` because I wanted to first ask if this is a reasonable way to do it or if there is a better way (@pukkandan)? Also: Do I need to add a test for each reseller?

Fixes #4630

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

### CODE REVIEW

**Comment by pukkandan** [comment]
It looks like `ZattooBaseIE._real_extract` is no longer needed. Other than that, it looks fine

> Do I need to add a test for each reseller?

Ideally, yes. But if that is difficult, just add for the salttv extractors

**Comment by goggle** [comment]
I don't see a `ZattooBaseIE._real_extract` in my code. But if you mean `ZattooPlatformBaseIE._real_extract`, it doesn't work without that method.

I have now adapted all the other resellers as well and tests have been added.

File: yt_dlp/extractor/zattoo.py [hunk 1]
```diff
@@ -327,139 +319,547 @@ class ZattooRecordingsIE(ZattooBaseIE):
     }]
 
 
-class NetPlusIE(ZattooPlatformBaseIE):
-    _NETRC_MACHINE = 'netplus'
+class NetPlusTVBaseIE(ZattooPlatformBaseIE):
+    _NETRC_MACHINE = 'netplustv'
```
Comment (pukkandan) [inline yt_dlp/extractor/zattoo.py:323]: Don't change netrc of existing extractor
```suggestion
    _NETRC = 'netplus'
```

**Review by pukkandan — APPROVED** [review]

File: yt_dlp/extractor/zattoo.py (hunk 1 above)
Comment (goggle) [reply yt_dlp/extractor/zattoo.py]: It used to be `_NETRC_MACHINE = 'netplus'`, not `_NETRC = 'netplus'`, I think this causes the tests to fail...

Comment (pukkandan) [reply yt_dlp/extractor/zattoo.py]: Yeah, my mistake. I'll fix it

**Pull request merged**